File: Desktop/winter_quarter/research/doing/FEA_code/2D_Zijiang/myfun_comsol_load_constant.m
% Load constants
function [] = myfun_comsol_load_constant()

global rt_electrode1X rt_electrode2X rt_electrode3X rt_electrode1Y rt_electrode2Y rt_electrode3Y ... % electrode coordinates
    AppliedVoltage InitialVoltage MatrixConductivity InterfaceConductivity CurrentSource FillerConductivity ... % electric properties
    FillerRelPerm InterfaceRelPerm InterfaceImagPerm ElectrodeConductivity ElectrodeRelPerm ...
    FreqRange TimeRange ACMode extra_E_infi

switch (ACMode)
    case (1)
    otherwise % DC mode. rectangular box
        rt_electrode1X                 = 0.125/0.3;
        rt_electrode1Y                 = 0;
        rt_electrode2X                 = 0.15/0.3;
        rt_electrode2Y                 = 0.03/0.15;
        rt_electrode3X                 = 0.175/0.3;
        rt_electrode3Y                 = 0;
end

% Model parameters
extra_E_infi                        = -0.55;                % polymer matrix adjustment: vertical shift from prony series output and expt data comparison
AppliedVoltage                      = 1.5e-3;               % Initial voltage applied on electrode
InitialVoltage                      = 0.01e-3;              % Initial voltage applied over entire domain
CurrentSource                       = 2e3;                   % Current source from electrode
FreqRange                           = 10.^(-3:0.2:6);       % Frequency range of AC terminal
TimeRange                           = 'range(0,0.25,20)';    % Time range for DC study

% Material Properties
MatrixConductivity                  = 1e-15;                 % Electric conductivity of fillers, matrix and interface
FillerConductivity                  = MatrixConductivity;    % Conductivity of filler
FillerRelPerm                       = 3.9;                   % 3x of matrix's. Relative permittivity of filler
InterfaceRelPerm                    = 2.4;                   % Relative permittivity of IF1
InterfaceImagPerm                   = 0.001; 
InterfaceConductivity               = 1e-11;                 % Conductivity of IF1
ElectrodeRelPerm                    = 2; 
ElectrodeConductivity               = 1e6;                  % Conductivity of electrode

File: Desktop/winter_quarter/research/doing/FEA_code/2D_java_code/myfun_comsol_load_constant.java
public class myfun_comsol_load_constant {
	private float rt_electrode1X;
	private float rt_electrode2X;
	private float rt_electrode3X;
	private float rt_electrode1Y;
	private float rt_electrode2Y;
	private float rt_electrode3Y;
	private float AppliedVoltage;
	private float InitialVoltage;
	private float MatrixConductivity;
	private float InterfaceConductivity;
	private float InterfaceRelPerm;
	private float InterfaceImagPerm;
	private float CurrentSource;
	private float FillerConductivity;
	private float FillerRelPerm;
	private float floaterfaceRelPerm;
	private float floaterfaceImagPerm;
	private float ElectrodeConductivity;
	private float ElectrodeRelPerm;
	private float FreqRange;
	private String TimeRange;
	private int ACMode;
	private int Compfreq;
	
	public static void myfun_comsol_load_constant() {
		if (ACMode == 1) {
		} else {
			rt_electrode1X = 0.125/0.3;
			rt_electrode1Y = 0;
			rt_electrode2X = 0.15/0.3;
			rt_electrode2Y = 0.03/0.15;
			rt_electrode3X = 0;
			rt_electrode3Y = 0;
		}
		
		AppliedVoltage = 1.5e-3;
		InitialVoltage = 0.01e-3;
		MatrixConductivity = 1e-15;
		CurrentSource = 2e3;
		FillerConductivity = MatrixConductivity;
		FillerRelPerm = 3.9;
		InterfaceRelPerm = 2.4;
		InterfaceImagPerm = 0.001;
		InterfaceConductivity = 1e-11;
		ElectrodeRelPerm = 2;
		ElectrodeConductivity = 1e6;
		FreqRange = 10.0^(-3:0.2:6); // what is this?
		TimeRange = 'range(0,0.25,20)';
	}
}